src/elements/stream_element.ts

```diff
@@ -36,7 +36,7 @@ export class StreamElement extends HTMLElement {
   }
 
   get duplicateChildren() {
-    return [...this.templateContent?.children].map(templateChild => {
+    return [...this.templateContent?.children].filter(templateChild => !!templateChild.id).map(templateChild => {
       let targetChild = [...this.targetElement!.children].filter(c => c.id === templateChild.id)[0]
       return { targetChild , templateChild }
```

Comment:
@jeromecornet 

I'd like to suggest a slight adjustment.  Since the method is "find duplicate children in the TARGET with the same ID as the TEMPLATE", I think the code should filter the TEMPLATE children with id's.

```diff
get duplicateChildren() {<br class="Apple-interchange-newline">
  return [...this.templateContent?.children].map(templateChild => {
-   let targetChild = [...this.targetElement!.children].filter(c => c.id === templateChild.id)[0]
+   let targetChild = [...this.targetElement!.children].filter(c => c.hasAttribute('id') && c.id === templateChild.id)[0]
    return { targetChild , templateChild }
```
It's also a bit more efficient (less loops).

Reply:
> I think the code should filter the TARGET children with id's.
> It's also a bit more efficient (less loops)
Either I don't understand your comment, or I simply don't agree (or both 😄)

the bit I added: `.filter(templateChild => !!templateChild.id)` runs before the `map`, so it ensures that the inner loop is only executed for the ids that exist in the template. So there are less loops in my version than in the ones you suggest, no ?

In my code, the only way `c.id === templateChild.id` in the inner `filter` is that `c.id` is not empty (since the filter ensures that `templateChild.id` is not empty), so the added test ` c.hasAttribute('id')` is redundant.

Am I missing something ?

Reply:
Clarification: "less loops" assuming the `this.targetElement` has more children than `this. templateContent`.  Otherwise, if the inverse I can see why we might filter by target.

One additional efficiency I noticed the code generates `[...this.targetElement!.children]` array for every templateChild in the loop.  It could be created once before the looping over `this.templateContent?.children`.

Reply:
@jeromecornet 

I recognize the redundancy of both changes.  I was suggesting filtering the targetChildren without ids instead of filtering out the templateChildren without ids.  I was not suggesting we do both.  They both accomplish the same result, it was feedback. based on code comprehension/readability and not functionality.

FYI, I started my code review prior to being merged.

Reply:
Was just working on this code in the branch to add css selectors for multiple replaces, and came up with this that I found much easier to follow/read:

```js
  removeDuplicateTargetChildren() {
    this.duplicateChildren.forEach(c => c.remove())
  }
  
  get duplicateChildren() {
    const existingChildren = this.targetElements.flatMap(e => [...e.children]).filter(c => !!c.id)
    const newChildrenIds   = [...this.templateContent?.children].filter(c => !!c.id).map(c => c.id)
  
    return existingChildren.filter(c => newChildrenIds.includes(c.id))
  }
```

Have not considered performance ramifications, though.

Reply:
The targetElements is from the PR that allows CSS selectors for targets.

Reply:
I like the suggested changes.